### PR TITLE
[Prototype] JSON Application API

### PR DESCRIPTION
**Pull request by plauche**
This PR sketches out an initial Application API:
- First pass of API functions needed to satisfy basic IPFS-in-space scenarios
- Currently the API is serialized to JSON for ease of debugging
- An `app-api-cli` tool for easy sending api commands

### CODE REVIEW

**Comment by ddimaria** [comment]
Can you add a README?

File: block-streamer/src/control.rs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+use crate::api::ApplicationAPI;
+use crate::receive::receive;
+use crate::transmit::transmit;
+
+pub async fn control(listen_addr: &String) -> Result<()> {
+    info!("Listening for control messages on {}", listen_addr);
+    let listen_address: SocketAddr = listen_addr.parse()?;
+
+    let mut buf = vec![0; 10240];
```
Comment (ddimaria) [inline block-streamer/src/control.rs:19]: Can you make a comment on the selection of the `10240` value?  You could make it an expressive constant to achieve this explanation as well.

File: block-streamer/src/control.rs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        if let Ok(msg) = from_slice::<ApplicationAPI>(&buf[..real_len]) {
+            match msg {
+                ApplicationAPI::Receive { path, listen_addr } => {
+                    receive(&PathBuf::from(path), &listen_addr).await?
+                }
+                ApplicationAPI::Transmit { path, target_addr } => {
+                    transmit(&PathBuf::from(path), &target_addr).await?
+                }
+                other => {
+                    info!("Received {:?}", other);
```
Comment (ddimaria) [inline block-streamer/src/control.rs:45]: Is this an error case?

File: block-streamer/src/control.rs (hunk 1 above)
Comment (plauche) [reply block-streamer/src/control.rs:19]: I'll address this in #20

File: block-streamer/src/control.rs (hunk 2 above)
Comment (plauche) [reply block-streamer/src/control.rs:45]: I'll address this in #20

**Pull request merged**